piet/src/text.rs

```diff
@@ -306,17 +306,24 @@ pub struct LineMetric {
     /// Includes trailing whitespace.
     pub end_offset: usize,
 
-    /// Length in (in UTF-8 code units) of current line's trailing whitespace.
+    /// The length of the trailing whitespace at the end of this line, in utf-8 code units.
     pub trailing_whitespace: usize,
 
-    /// Distance of the baseline from the top of the line
+    /// The distance from the top of the line (`y_offset`) to the baseline.
     pub baseline: f64,
 
-    /// Line height
+    /// The height of the line.
     pub height: f64,
 
     /// Cumulative line height (includes previous line heights)
+    #[deprecated(since = "0.2.0", note = "use y_offset instead")]
     pub cumulative_height: f64,
+
+    /// The y position of the top of this line, relative to the top of the layout.
```

Comment:
Is it an invariant that y_offset of the next line is equal to y_offset of this line + height? If so, it should be stated. If not, the reason why not should be clarified. (Obviously this depends on the way line gap is handled)

Reply:
It's not, although it is true in many cases. There are fonts that this does not hold for, though (again, Courier on mac) and I haven't investigated.

I think making these firm invariants will only be possible when we have finer grained control over layout and line spacing.